### ISSUE TITLE
Non-ASCII symbols in a shortcut name/descr. cause errors 

### ISSUE THREAD

**Issue by gornostal** [Ulauncher/Ulauncher#82]
```
2017-02-26 18:27:43,437:ERROR:ulauncher: except_hook() 'Uncaught exception'
Traceback (most recent call last):
  File "/home/agornostal/projects/ulauncher/ulauncher/ui/windows/UlauncherWindow.py", line 144, in on_input_changed
    Search.get_instance().start(self.get_user_query())
  File "/home/agornostal/projects/ulauncher/ulauncher/search/__init__.py", line 22, in start
    self.choose_search_mode(query).on_query(query).run_all()
  File "/home/agornostal/projects/ulauncher/ulauncher/search/DefaultSearchMode.py", line 23, in on_query
    result_list.extend(shortcut_items)
  File "/home/agornostal/projects/ulauncher/ulauncher/search/SortedResultList.py", line 42, in extend
    map(lambda i: self.append(i), items)
  File "/home/agornostal/projects/ulauncher/ulauncher/search/SortedResultList.py", line 42, in <lambda>
    map(lambda i: self.append(i), items)
  File "/home/agornostal/projects/ulauncher/ulauncher/search/SortedResultList.py", line 45, in append
    score = get_score(self._query, result_item.get_name())
  File "/home/agornostal/projects/ulauncher/ulauncher/utils/fuzzy_search.py", line 64, in get_score
    if text_part.startswith(query):
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd1 in position 0: ordinal not in range(128)
```

**Comment by mbirth** [comment]
This seems still to happen in 4.0.6.r1 (on Ubuntu 17.10) but in another part:

```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib/python2.7/dist-packages/ulauncher/search/apps/app_watcher.py", line 136, in start
    added_apps = map(lambda app: db.put_app(app), find_apps())
  File "/usr/lib/python2.7/dist-packages/ulauncher/util/desktop/reader.py", line 56, in find_apps
    return filter(filter_app, map(read_desktop_file, find_desktop_files(dirs)))
  File "/usr/lib/python2.7/dist-packages/ulauncher/util/desktop/reader.py", line 24, in find_desktop_files
    if any([file.startswith(dir) for dir in blacklisted_dirs]):
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 72: ordinal not in range(128)
```

**Comment by ulukyn** [comment]
Easy fix for the issue (works for me)
file : ulauncher/util/desktop/reader.py line 24

```
blacklisted_dirs_srt = Settings.get_instance().get_property('blacklisted-desktop-dirs')
blacklisted_dirs = blacklisted_dirs_srt.split(':') if blacklisted_dirs_srt else []
for file in files:
   >>> file = file.decode("utf-8") <<<
   if any([file.startswith(dir) for dir in blacklisted_dirs]):
       continue
```

**Comment by mbirth** [comment]
I fixed it differently for me (same file):

```python
from ulauncher.util.string import force_unicode
# <...>
    for file in files:
        if any([force_unicode(file).startswith(dir) for dir in blacklisted_dirs]):
            continue
```

**Comment by gornostal** [comment]
Thanks for the fix :+1:

**Comment by gornostal** [comment]
The was released with Ulauncher v4.0.7.r5
https://github.com/Ulauncher/Ulauncher/releases/tag/4.0.7.r5

Please reopen if needed.
